File: backend/src/main/java/com/example/projectPfe/services/Interface/AssuranceService.java
package com.example.projectPfe.Services.Interface;

import com.example.projectPfe.models.Assurance;

import java.util.List;
import java.util.Optional;

public interface AssuranceService {

    List<Assurance> getAllAssurances();
    Assurance getAssuranceById(int id);
    Assurance createAssurance(Assurance assurance);
    Assurance updateAssurance(int id, Assurance assurance);
    void deleteAssurance(int id);

}

File: front_end/src/app/Services/assurance.service.ts
import { Injectable } from '@angular/core';
import { Assurance } from '../Models/Assurance';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AssuranceService {
  private baseUrl = 'http://localhost:8034/api/assurance';

  constructor(private http: HttpClient) { }

  getAllAssurances(): Observable<Assurance[]> {
    return this.http.get<Assurance[]>(this.baseUrl);
  }

  getAssuranceById(id: number): Observable<Assurance> {
    return this.http.get<Assurance>(`${this.baseUrl}/${id}`);
  }

  createAssurance(assurance: Assurance): Observable<Assurance> {
    return this.http.post<Assurance>(this.baseUrl, assurance);
  }

  updateAssurance(id: number, assurance: Assurance): Observable<Assurance> {
    return this.http.put<Assurance>(`${this.baseUrl}/${id}`, assurance);
  }

  deleteAssurance(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
  
}
